src/services/conversations/conversations.hooks.js

```diff
@@ -0,0 +1,165 @@
+import logger from 'winston';
+import commons from 'feathers-hooks-common';
+import { updatedAt, createdAt } from '../../hooks/timestamps';
+import { disallow } from 'feathers-hooks-common';
+import sanitizeAddress from '../../hooks/sanitizeAddress';
+import errors from 'feathers-errors';
+import sanitizeHtml from '../../hooks/sanitizeHtml';
+
+/**
+  API for creating conversations
+  The following params are accepted when creating a message for a conversation
+
+  @params:
+    milestoneId:        (string) the id of the milestone that this conversation belongs to
+    messageContext:     (string) context of the message, see MESSAGE_CONTEXT below
+    message:            (string) the actual message
+    replyToId:          (string) id of the message that this is a reply to
+
+  @param ownerAddress (string) is automatically set based on the current logged in user
+**/
+
+
+/**
+  Available conversation types. This roughly follows the milestone status
+  replyTo is for threaded messages
+**/
+const MESSAGE_CONTEXT = [
+  'proposed',
+  'rejected',
+  'NeedsReview',
+  'Completed',
+  'Canceled',
+  'replyTo',
+  'proposedRejected',
+  'proposedAccepted',
+  'rePropose'
+];
+
+/**
+ Only people involved with the milestone can create conversation
+ Sets the address creating the conversation as the owner
+ */
+const restrictAndSetOwner = () => context => {
+  const { app, params } = context;
+  const { milestoneId } = context.data;
+  let { user } = params;
+
+  // external calls need a user
+  if (!user && context.params.provider) throw new errors.NotAuthenticated();
+
+  return new Promise((resolve, reject) => {
+    app
+      .service('milestones')
+      .get(milestoneId)
+      .then(milestone => {
+        // for internal calls there's no user, so the user creating the message is stored on the milestone
+        // for external calls, the currentuser creates the message
+        context.data.ownerAddress = user && user.address || milestone.performedByAddress;
+        delete context.data.user;
```

Comment:
it looks like this used in feathers? can we just not set it in the dapp?

Reply:
I don't think so because the message is (not always) created from the dapp, it is created by feathers after a milestone tx is mined. `ownerAddress` is kept on the message as a reference who created the message.

But maybe I don't understand the question?

Reply:
I was talking about the line `delete context.data.user;` data.user doesn't appear to be used anywhere, and we delete it here

Reply:
You're right, legacy code. Removed it